index.js

```diff
@@ -60,10 +60,11 @@ function compression (options) {
     var length
     var listeners = []
     var stream
+    var noop = function () {}
 
-    var _end = res.end
+    res._end = res.end
```

Comment:
You cannot store the references as res._end  and _write. There is no code where that check nothing is being overwritten, for example. Using this module twice will no longer function, etc.

Doing this causes too many bugs to be able to accept. Why did you have to change this?